### ISSUE TITLE
Empty chart displayed for virtual activities

### ISSUE THREAD

**Issue by lordvalium** [ianperrin/MMM-Strava#61]
When submitting a new issue, please supply the following information:

Platform (Hardware/OS):
Raspberry Pi 4

Node Version:
6.14.15

MagicMirror Version:
v2.16.0

Description: Virtual Rides are are not aggregated with Rides

Steps to Reproduce: Install module, have some virtual rides and rides in the own activity feed. They wont be shown together if i select "Ride" as single category

Expected Results: Combined stats (virtual ride + ride)

Actual Results: just ride

Configuration: 
````
    {
      module: 'MMM-Strava',
      position: 'top_right',
      header: "phil' strava",
      config: {
        client_id: '******',
        client_secret: '*****************',
        activities: [
          'ride',
          'run',
          'hike'
        ],
        mode: 'chart',
        period: 'ytd'
      }
    },
````
Additional Notes: 
I saw that issue https://github.com/ianperrin/MMM-Strava/issues/31
Does that indicate the same problem? After some research i found some _replace_ code in node_helpers. But it dont work yet.

**Comment by ianperrin** [comment]
Hi @lordvalium - thanks for raising the issue. 

Virtual activities should be merged with their non-virtual counterparts. However, you need to add them to the activities list in the config so that the data can be queried from the Strava api.

Can you try amending your config as follows as let me know whether this works for you:

````
...
        activities: [
          'ride',
          'virtualride',
          'run',
          'hike'
        ],
...
````

**Comment by lordvalium** [comment]
hi

Thank you for your reply.
Unfortunately that does not work 

Gruß 


Am 08.09.2021 um 17:13 schrieb Ian ***@***.***>:

﻿
Hi @lordvalium - thanks for raising the issue.

Virtual activities should be merged with their non-virtual counterparts. However, you need to add them to the activities list in the config so that the data can be queried from the Strava api.

Can you try amending your config as follows as let me know whether this works for you:

...
        activities: [
          'ride',
          'virtualride',
          'run',
          'hike'
        ],
...

**Comment by ianperrin** [comment]
> Unfortunately that does not work

Can you describe (and possibly provide screenshots) for what is displayed using the following configurations?

1. `activities: ["ride"],`
1. `activities: ["virtualride"],`
1. `activities: ["ride", "virtualride"],`

Thanks

**Comment by lordvalium** [comment]
that is the config
![image](https://user-images.githubusercontent.com/2970225/132576622-34eb7aa9-6cf1-47fd-9c93-93a65492d41c.png)
 
but i have nearly 4000km in sum. 
![image](https://user-images.githubusercontent.com/2970225/132576803-135f6b84-20e7-40b7-9a0b-3a2c11cdb3eb.png)


also virtualride is not shown properly, as it shows 0

**Comment by ianperrin** [comment]
Hi @lordvalium 

Thanks for the screenshots. I think you are experiencing a couple of issues.

Firstly, the chart module is only summarising the first 200 activities returned from the API for the selected period (see issue #46). Any additional activities are ignored. I'm working on a fix for this at the moment. Looking at your stats, I suspect you may have run into this issue. 

Secondly, the module template is still generating a chart for `virtualride` activities, which contains no data because the module has aggregated them together with `ride` activities. This is a bug which I was unaware of until now so will work on a fix.

A temporary workaround whilst I amend the template and work on a solution for athletes with more than 200 activities in given period is to change the config as follows...
````
    period: 'recent',
    activities: [ 'ride', 'virtualride', 'run', 'hike' ],
```` 

This will (hopefully) keep you within the 200 activity limit, but will still show the empty `virtualride` chart. To hide this, add the following to your custom.css file
````
 .MMM-Strava div#virtualride {
   display: none;
 }
````

**Comment by lordvalium** [comment]
cool, that worked, so the empty is not displayed anymore and recent works. 

Thanks for keeping an eye on if for the ytd view. 

Thats the nice number to be shown on the screen and motivating :P 

cheers

**Comment by albertusQuelle** [comment]
Now the virtual is accumulating. But everything after (eg hike and/or run) disappeared 

Cheers 

Am 10.09.2021 um 17:21 schrieb Ian ***@***.***>:

﻿
Closed #61 via fc9f57a.

**Comment by ianperrin** [comment]
Hi @lordvalium - that's curious. Using a test account with activities of type ride, virtualride, run and hike, I can't seem to  replicate this problem.

Here's a screenshot showing this working with that account when using `activities: [ 'ride', 'virtualride', 'run', 'hike' ]`

![image](https://user-images.githubusercontent.com/2277491/132951405-46f52207-4321-486a-b952-2fdabbb82b46.png)

Here's a couple of thoughts
* Can you remove the changes to the custom.css file
* Can you double check the config you are using to make sure all the activity types you require are included in the `activities` option
* Can you make sure you have the latest version and re-install the dependencies as described in the readme - https://github.com/ianperrin/MMM-Strava#updating-the-module including `npm install --production`? I published some updates for the module yesterday.

Thanks
